spec/client_spec.rb

```diff
@@ -258,32 +288,6 @@ def redis.with
       end
     end
 
-    context 'when a server goes away' do
-      it 'raises an error on connection issues' do
-        # Set lock manager to a (hopefully) non-existent Redis URL to test error
-        redis_instance = lock_manager.instance_variable_get(:@servers).first
-        redis_instance.instance_variable_set(:@redis, unreachable_redis)
-
-        expect {
-          lock_manager.lock(resource_key, ttl)
-        }.to raise_error(RedisClient::CannotConnectError)
```

Comment:
I think you should leave these tests and change them to pass based on updated logic, e.g. change two instances to broken state and assert that there's an error

Reply:
The `raises an error on connection issues` test is pretty much identical to `fails to acquire a lock if majority of redis instances are not available` so I left it out. Re-added and refactored `recovers from connection issues`test.